### ISSUE TITLE
How to make front view dim when sidemenu is shown?

### ISSUE THREAD

**Issue by dtran26** [John-Lluch/SWRevealViewController#719]
Hi,

I want to achieve a gradual visual effect of a dark screen on the front view when sidemenu is shown. Doing my coding in swift 3.0. Any suggestions?

Thanks

**Comment by iDevelopper** [comment]
What do you mean by a gradual visual effect? Have you a screenshot?

**Comment by dtran26** [comment]
![img_5937](https://user-images.githubusercontent.com/7690403/29109204-0ef16430-7c97-11e7-9b25-ca011a46b5a2.PNG)
![img_5936](https://user-images.githubusercontent.com/7690403/29109207-159f05bc-7c97-11e7-9343-43f1d9371e59.PNG)

**Comment by iDevelopper** [comment]
This is just a view with an alpha < 1 which is added on the front view controller view when menu is opening...

**Comment by dtran26** [comment]
Yes but I was wondering how do I achieve a slow effect with gesture recognizer (i.e. the more frontview is hidden the more view's alpha is). Sorry if this is a silly question I started iOS recently

**Comment by iDevelopper** [comment]
You can implement the delegate methods and check for progress argument:

```
func revealController(_ revealController: SWRevealViewController!, panGestureMovedToLocation location: CGFloat, progress: CGFloat, overProgress: CGFloat)

func revealController(_ revealController: SWRevealViewController!, willMoveTo position: FrontViewPosition) 
```

Sample:

[SWHomePushSwift 2.zip](https://github.com/John-Lluch/SWRevealViewController/files/1213751/SWHomePushSwift.2.zip)

**Comment by dtran26** [comment]
@iDevelopper Thank you very much!!
